test/react-markdown.test.js

```diff
@@ -893,3 +893,10 @@ test('should support formatting at the start of a GFM tasklist (GH-494)', () =>
   const expected = '<ul><li><input type="checkbox" readonly=""/><em>a</em></li></ul>'
   expect(actual).toEqual(expected)
 })
+
+test('should not crush on ending tag at start', () => {
+  const input = '<ruby><ruby></ruby></ruby>'
+  expect(() =>
+    renderHTML(<Markdown source={input} escapeHtml={false} astPlugins={[htmlParser()]} />)
+  ).not.toThrow()
```

Comment:
@cerkiewny what is the actual resulting value?
Could that be included in the test?

Reply:
to be honest I expect it just not to throw, I can add expectation that it is returning whatever it is returning now (which is <p><ruby></ruby><ruby></ruby></p>) but for me it isn't really obvious this is expected, let me know if I should add it to the test or leave it as is. To phrase my concerns otherwise I think even if it would change to something like <p><ruby></ruby></p> I don't think this test should stop passing.

I am happy to add it to the test though if you think it is appropriate change.

Reply:
I ask because `remark` and it's related project aim to match commonmark.
https://spec.commonmark.org/dingus/?text=%3Cruby%3E%3Cruby%3E%3C%2Fruby%3E%3C%2Fruby%3E%0A
which for this example would produce `<p><ruby><ruby></ruby></ruby></p>`

Reply:
I understand, the thing is, for now it produces the exception. And my fix was meant to just mitigate just that. I am not sure how to proceed, I can dive deeper into the parser and try and understand and fix the expectation mismatch however it will probably take a while. Meanwhile the output doesn't match expectation and the markdown parser simply throws. So we can either wait for the fix or merge as-is with follow up to fix the expectation. What do you think?